### PR TITLE
Filter out expired entities in JpaMapKeycloakTransaction

### LINKED ISSUE #12623: Add explicit expiration check to each query for ExpirableEntities for JPA store
Follow up on #12563 for JPA  store

### PR DESCRIPTION
**Pull request by sguilhen**
Closes #12623

### CODE REVIEW

**Comment by sguilhen** [comment]
@mhajas This is basically the same idea you came up with earlier, with some minor adjustments. I've also checked the other expirable entities and they all are following the same pattern of using a column named `expiration`.

I've done some manual testing using auth events, setting expiration to 1 minute in the console and then updating the search after 1 minute to check the events are no longer displayed. Seems to be working according to the plan.

**Review by mhajas — APPROVED** [review]
Thanks @sguilhen! I thought it will be more complicated. I should include this in my PR and not create a new issue for this, sorry.

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java [hunk 1]
```diff
@@ -98,7 +103,8 @@ public E read(String key) {
         if (key == null) return null;
         UUID uuid = StringKeyConverter.UUIDKey.INSTANCE.fromStringSafe(key);
         if (uuid == null) return null;
-        return mapToEntityDelegateUnique(em.find(entityType, uuid));
+        E e = mapToEntityDelegateUnique(em.find(entityType, uuid));
+        return e != null && isExpirableEntity && isExpired((ExpirableEntity) e, true) ? null : e;
```
Comment (vramik) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java:107]: If I got it right we should never obtain an entity which is actually expired, right? So this check is here to be sure, is that so?

**Review by vramik — COMMENTED** [review]
Thank you @sguilhen. Changes looks good to me. I was wondering whether it might make sense to also create model to verify that we actually don't return an expired entity, wdyt?

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java (hunk 1 above)
Comment (mhajas) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapKeycloakTransaction.java:107]: Yes, you are right. Ideally, Keycloak should not know about the presence of expired entities in the store.

**Review by hmlnarik — APPROVED** [review]
Thank you for the PR. Approving per @mhajas 's approval

**Pull request merged**